src/components/results/detail/helpers/DocumentViewer.vue

```diff
@@ -0,0 +1,126 @@
+<template>
+  <div>
+    <v-alert
+      border="left"
+      color="red lighten-2"
+      v-if="error"
+    >
+      <i>Error loading preview</i>
+    </v-alert>
+    <v-alert
+      border="left"
+      color="blue lighten-4"
+      v-else-if="!srcUrl"
+    >
+      <i>Loading preview</i>
+      <v-progress-linear
+        :indeterminate="progress===0"
+        :value="$data.progress"
+      />
+    </v-alert>
+    <iframe
+      v-else
+      :src="srcUrl"
+      width="100%"
+      height="700"
+      :type="file.mimetype"
+    />
+  </div>
+</template>
+
+<script>
+
+import mime from 'mime';
+import GoldenRetriever from '@/helpers/GoldenRetriever';
+// import { getFileExtension } from '@/helpers/fileHelper';
+import getResourceURL from '@/helpers/resourceURL';
+
+export default {
+  created() {
+    this.fetch();
+  },
+  data() {
+    return {
+      error: false,
+      progress: 0,
+      srcUrlFromBlob: '',
+    };
+  },
+  props: {
+    file: {
+      type: Object,
+      required: true,
+    },
+    active: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  computed: {
+    srcUrl() {
+      // console.log(this.$props.file.mimetype, mime.getExtension(this.$props.file.mimetype));
+      switch (mime.getExtension(this.$props.file.mimetype)) {
+        case 'epub':
+          return `https://readium.web.app/?epub=${getResourceURL(this.file.hash)}`;
+        // case 'rtf': // rtf does not work for some reason
+        case 'docx':
+        case 'xlsx':
+        case 'odf':
+        case 'odt':
+          // return `https://docs.google.com/gview?embedded=true&url=${
```

Comment:
It might be good to add a comment stating why we opted for the Office viewer, for our future selves (and others). Perhaps this one: https://stackoverflow.com/questions/36592508/cannot-embed-doc-docx-files-in-google-doc-viewer

Reply:
I believe this just answers, why _not_ using google doc viewer. BTW, isn't there a nice LibreOffice online viewer or sth? Fits more the FOSS philosophy.

Reply:
for odf, there is this: (didn't try it) https://odfviewer.nsspot.net/

Reply:
The one in the original ticket is fine. I was wondering why you didn't use that. If possible I would prefer that one. https://github.com/ipfs-search/dweb-search-frontend/issues/25

Reply:
1. viewerjs is an awesome project, but got abandoned 7 years ago. If you want to use that in stead, it is just adding 1 or 2 lines of code. (And you can drop the whole retriever stuff.)
2. office viewer is what i used
3. pdf.js is great but not trivial to implement. 

I think the version as it is now supports a lot of use cases, and save for some minor bugs, it will suffice.